osu.Game.Rulesets.Catch/Objects/Drawables/CaughtObjectContainer.cs

```diff
@@ -0,0 +1,334 @@
+                caughtDropletPool = new DrawablePool<DrawableCaughtDroplet>(25),
+                StackedObjectContainer = new Container<DrawableCaughtObject>()
+            };
+
+            lifetimeManager.EntryBecameAlive += entryBecameAlive;
+            lifetimeManager.EntryBecameDead += entryBecameDead;
+        }
+
+        /// <summary>
+        /// Add a caught object to the stack.
+        /// </summary>
+        public CaughtObjectEntry AddStackObject(IHasCatchObjectState source, Vector2 positionInStack)
+        {
+            var delayedDropEntry = new DroppedObjectEntry(positionInStack, source);
+            var stackEntry = new StackedObjectEntry(positionInStack, delayedDropEntry, source);
+
+            addImmediateEntry(stackEntry);
+
+            // `DropStackedObjects` may be called before lifetime update.
+            if (stackEntry.LifetimeStart <= Clock.CurrentTime)
+                aliveStackedObjects.Add(stackEntry);
+
+            randomSeed++;
+
+            return stackEntry;
+        }
+
+        /// <summary>
+        /// Immediately drop a caught object.
+        /// </summary>
+        public CaughtObjectEntry AddDropObject(IHasCatchObjectState source, Vector2 positionInStack, DroppedObjectAnimation animation, int mirrorDirection)
+        {
+            var dropEntry = new DroppedObjectEntry(positionInStack, source)
+            {
+                Animation = animation,
+                DropPosition = getCurrentDropPosition(positionInStack),
+                MirrorDirection = mirrorDirection
+            };
+
+            addImmediateEntry(dropEntry);
+
+            randomSeed++;
+
+            return dropEntry;
+        }
+
+        /// <summary>
+        /// Remove caught objects
+        /// </summary>
+        public bool RemoveEntry(CaughtObjectEntry entry)
+        {
+            randomSeed--;
+
+            return removeImmediateEntry(entry);
+        }
+
+        public Vector2 GetPositionInStack(Vector2 position, float displayRadius)
+        {
+            const float radius_div_2 = CatchHitObject.OBJECT_RADIUS / 2;
+            const float allowance = 10;
+
+            int iteration = 0;
+
+            while (aliveStackedObjects.Any(f => Vector2Extensions.Distance(f.PositionInStack, position) < (displayRadius + radius_div_2) / (allowance / 2)))
+            {
+                float diff = (displayRadius + radius_div_2) / allowance;
+
+                position.X += (StatelessRNG.NextSingle(randomSeed, iteration * 2) - 0.5f) * diff * 2;
+                position.Y -= StatelessRNG.NextSingle(randomSeed, iteration * 2 + 1) * diff;
+
+                iteration++;
+            }
+
+            position.X = Math.Clamp(position.X, -CatcherArea.CATCHER_SIZE / 2, CatcherArea.CATCHER_SIZE / 2);
+
+            return position;
+        }
+
+        public void DropStackedObjects(DroppedObjectAnimation animation, int mirrorDirection)
```

Comment:
nit: would unify up naming here (there's `AddStackObject` and `AddDropObject`, but `DropStackedObject`).